packages/fx-core/src/component/driver/aad/utility/aadAppClient.ts

```diff
@@ -84,20 +85,31 @@ export class AadAppClient {
     return response.data.secretText;
   }
 
-  public async updateAadApp(manifest: AADManifest): Promise<void> {
+  public async updateAadApp(manifest: AADManifest): Promise<AxiosResponse<any>> {
     const objectId = manifest.id!; // You need to ensure the object id exists in manifest
     const requestBody = AadManifestHelper.manifestToApplication(manifest);
-    await this.axios.patch(`applications/${objectId}`, requestBody, {
-      "axios-retry": {
-        retries: this.retryNumber,
-        retryDelay: axiosRetry.exponentialDelay,
-        retryCondition: (error) =>
-          axiosRetry.isNetworkError(error) ||
-          axiosRetry.isRetryableError(error) ||
-          this.is404Error(error) || // also retry 404 error since AAD need sometime to sync created AAD app data
-          this.is400Error(error), // sometimes AAD will complain OAuth permission not found if we pre-authorize a newly created permission
-      },
-    });
+    try {
+      const response = await this.axios.patch(`applications/${objectId}`, requestBody, {
+        "axios-retry": {
+          retries: this.retryNumber,
+          retryDelay: axiosRetry.exponentialDelay,
+          retryCondition: (error) =>
+            axiosRetry.isNetworkError(error) ||
+            axiosRetry.isRetryableError(error) ||
+            this.is404Error(error) || // also retry 404 error since AAD need sometime to sync created AAD app data
+            this.is400Error(error), // sometimes AAD will complain OAuth permission not found if we pre-authorize a newly created permission
+        },
+      });
+      if (
```

Comment:
Per existing error handling code, axios will throw an error when the status code is 4xx, so we need to movethese logic to the catch block?

Reply:
By the way, please add a test case for the new logic

Reply:
still working on this PR, not ready.